x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/package-info.java

```diff
@@ -99,31 +99,81 @@
  *     <li>{@link org.elasticsearch.compute.data.Block} - fundamental unit of data.  Operations vectorize over blocks.</li>
  *     <li>{@link org.elasticsearch.compute.data.Page} - Data is broken up into pages (which are collections of blocks) to
  *     manage size in memory</li>
  * </ul>
- * </li>
- * <li>{@link org.elasticsearch.xpack.esql.core} - Core Utility Classes
+ *
+ * <h3>Core Classes</h3>
+ * {@link org.elasticsearch.xpack.esql.core} - Core Utility Classes
  * <ul>
+ *     <li>{@link org.elasticsearch.xpack.esql.session.EsqlSession} - Manages state across a query</li>
  *     <li>{@link org.elasticsearch.xpack.esql.core.type.DataType} - ES|QL is a typed language, and all the supported data types
  *     are listed in this collection.</li>
  *     <li>{@link org.elasticsearch.xpack.esql.core.expression.Expression} - Expression is the basis for all functions in ES|QL,
  *     but see also {@link org.elasticsearch.xpack.esql.evaluator.mapper.EvaluatorMapper}</li>
+ *     <li>{@link org.elasticsearch.xpack.esql.expression.function.EsqlFunctionRegistry} - Resolves function names to
+ *         function implementations.</li>
+ *     <li>{@link org.elasticsearch.xpack.esql.action.RestEsqlQueryAction Sync} and
+ *         {@link org.elasticsearch.xpack.esql.action.RestEsqlAsyncQueryAction async} HTTP API entry points</li>
  * </ul>
- * </li>
- * <li>{@link org.elasticsearch.xpack.esql.session.EsqlSession} - Manages state across a query</li>
- * <li>{@link org.elasticsearch.xpack.esql.analysis.Analyzer} - The first step in query processing</li>
- * <li>{@link org.elasticsearch.xpack.esql.expression.function.EsqlFunctionRegistry} - Resolves function names to
- *     function implementations.</li>
- * <li>{@link org.elasticsearch.xpack.esql.optimizer.LogicalPlanOptimizer} - Coordinator level logical optimizations</li>
- * <li>{@link org.elasticsearch.xpack.esql.optimizer.LocalLogicalPlanOptimizer} - Data node level logical optimizations</li>
- * <li>{@link org.elasticsearch.xpack.esql.action.RestEsqlQueryAction Sync} and
- *     {@link org.elasticsearch.xpack.esql.action.RestEsqlAsyncQueryAction async} HTTP API entry points</li>
+ *
+ * <h3>Query Planner</h3>
+ * <p>The query planner encompasses a huge chunk of serving a query.  Essentially, this covers everything from the output of the Antlr
+ * parser through to the actual computations and lucene operations.</p>
+ * <p>Two key concepts in the planner layer:</p>
+ * <ul>
+ *     <li>Logical vs Physical optimization - Logical optimizations refer to things that can be done strictly based on the structure
+ *     of the query, while Physical optimizations take into account information about the index or indices the query will execute
+ *     against</li>
+ *     <li>Local vs non-local operations - "local" refers to operations happening on the data nodes, while non-local operations generally
+ *     happen on the coordinating node and can apply to all participating nodes in the query</li>
```

Comment:
I think I'd call them `coordinating node operations` rather than non-local. But whatever

Reply:
So the code mostly just says local or blank.  `LocalLogicalPlanOptimizer` and `LogicalPlanOptimizer`.  You kind of have to know that if it doesn't say "local" it means "coordinating node"